### DISCUSSION TITLE
How to change line number format based on normal/insert mode

### DISCUSSION THREAD

**Discussion by m-zheng** [JetBrains/ideavim#478] [Q&A] [answered]
Hi guys, I am trying to migrate the vim setting to ideavim. I would like to change line number format based on normal/insert mode. The setting for vim comes from [here](https://jeffkreeftmeijer.com/vim-number/#hybrid-line-numbers). Anyone know how to modify the setting below so that it can work in ideavim? Thanks.

```
:set number

:augroup numbertoggle
:  autocmd!
:  autocmd BufEnter,FocusGained,InsertLeave,WinEnter * if &nu && mode() != "i" | set rnu   | endif
:  autocmd BufLeave,FocusLost,InsertEnter,WinLeave   * if &nu                  | set nornu | endif
:augroup END
```

**Comment by citizenmatt** [comment, marked as answer]
Unfortunately, this isn't possible right now - IdeaVim doesn't support `autocmd`. You can vote or follow [VIM-1693](https://youtrack.jetbrains.com/issue/VIM-1693) for this issue.
